stock_picking_dispatch/tests/test_dispatch.py

```diff
@@ -0,0 +1,279 @@
+# -*- coding: utf-8 -*-
+
+from datetime import date, timedelta
+
+from openerp.exceptions import UserError, ValidationError
+from openerp.tests.common import TransactionCase
+
+
+class TestPickingDispatch(TransactionCase):
+
+    def setUp(self):
+        super(TestPickingDispatch, self).setUp()
+        self.user_demo = self.env.ref('base.user_demo')
+
+        self.dispatch_model = self.env['picking.dispatch']
+        self.picking_model = self.env['stock.picking']
+
+        self.picking = self.create_simple_picking([
+            self.ref('product.product_product_6'),
+            self.ref('product.product_product_7')
+        ])
+        self.picking.action_confirm()
+
+        self.dispatch = self.dispatch_model.create({
+            'picker_id': self.env.user.id,
+            'move_ids': [
+                (4, line.id) for line in self.picking.move_lines
+            ]
+        })
+
+    def create_simple_picking(self, product_ids):
+        stock_loc = self.ref('stock.stock_location_stock')
+        customer_loc = self.ref('stock.stock_location_customers')
+
+        return self.env['stock.picking'].create({
+            'picking_type_id': self.ref('stock.picking_type_out'),
+            'location_id': stock_loc,
+            'location_dest_id': customer_loc,
+            'move_lines': [
+                (0, 0, {
+                    'name': 'Test move',
+                    'product_id': product_id,
+                    'product_uom': self.ref('product.product_uom_unit'),
+                    'product_uom_qty': 1,
+                    'location_id': stock_loc,
+                    'location_dest_id': customer_loc
+                }) for product_id in product_ids
+            ]
+        })
+
+    def test_picking_search_related_dispatch(self):
+        pickings = self.picking_model.search([
+            ('related_dispatch_ids', '=', self.dispatch.name)
+        ])
+
+        self.assertEqual(pickings, self.picking)
+
+    def test_picking_related_dispatch(self):
+        # Coverage test
+        empty_picking = self.picking_model.new()
+        self.assertFalse(empty_picking.related_dispatch_ids)
+
+        self.assertEqual(self.dispatch, self.picking.related_dispatch_ids)
+
+    def test_constraint(self):
+        with self.assertRaises(ValidationError):
+            self.dispatch_model.create({'state': 'assigned'})
+        self.dispatch_model.create({
+            'picker_id': self.env.user.id, 'state': 'assigned'
```

Comment:
`self.env.uid` exists in environement, you can use it instead of `self.env.user.id`